File: LP2/selectionsort.cpp
#include <iostream>
using namespace std;

// Function to perform selection sort
void selectionsort(int arr[], int n) {
    int i, j, temp, min;
    for (i = 0; i < n - 1; i++) {
        min = i;
        for (j = i + 1; j < n; j++) {
            if (arr[j] < arr[min]) {
                min = j; // Update the index of the minimum element
            }
        }
        // Swap the elements
        temp = arr[i];
        arr[i] = arr[min];
        arr[min] = temp;
    }

    // Display sorted array in ascending order
    cout << "\nSORTED ARRAY IN ASCENDING ORDER:\n";
    for (i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << "\n";

    // Display sorted array in descending order
    cout << "\nSORTED ARRAY IN DESCENDING ORDER:\n";
    for (i = n - 1; i >= 0; i--) {
        cout << arr[i] << " ";
    }
    cout << "\n";
}

int main() {
    int n;
    cout << "Enter the size of the array:\n";
    cin >> n;

    // Edge case handling
    if (n <= 0) {
        cout << "Error: Array size must be positive!\n";
        return 1;
    }

    int arr[n];
    cout << "Enter the elements of the array:\n";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    // Perform selection sort
    selectionsort(arr, n);

    return 0;
}

File: LP-II/selectionsort.py
# Selection Sort in Python
# Time Complexity: O(n^2)

def selectionSort(array, size):
    for i in range(size):
        min_index = i

# Find the index of the minimum element in the unsorted part
    for j in range(i + 1, size):
        if array[j] < array[min_index]:
            min_index = j

# Swap the found minimum element with the first element
    array[i], array[min_index] = array[min_index],array[i]

# Driver Code
arr = [-2, 45, 0, 11, -9, 88, -97, -202, 747]
size = len(arr)

selectionSort(arr, size)

print("The array after sorting in Ascending Order by Selection Sort is:")
print(arr)
